sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.external.controller.searchStudent.e2e-spec.ts

```diff
@@ -352,6 +493,271 @@ describe("StudentExternalController(e2e)-searchStudents", () => {
+   * - `hasDependants` whether the student has dependants.
+   * - `isLivingAway` whether the student is living away.
+   * - `isMarried` whether the student is married.
+   * @returns application search data.
+   */
+  async function createStudentApplicationSearchData(
+    student: Student,
+    options?: {
+      hasDependants?: boolean;
+      isLivingAway?: boolean;
+      isMarried?: boolean;
+    },
+  ): Promise<Application> {
+    const fakeApplicationData: ApplicationData = { workflowName: "DUMMY" };
+    const fullTimeAssessmentData = {
+      tuitionCost: 600,
+      booksAndSuppliesCost: 500,
+      exceptionalEducationCost: 400,
+      livingAllowance: 300,
+      secondResidenceCost: 0,
+      childcareCost: 0,
+      alimonyOrChildSupport: 0,
+      transportationCost: 0,
+      provincialAssessmentNeed: 1800,
+    } as FullTimeAssessment;
+    const workflowData = {
+      studentData: {
+        citizenship: "canadianCitizen",
+        taxReturnIncome: 1000,
+        bcResident: FormYesNoOptions.Yes,
+        relationshipStatus: RelationshipStatus.Single,
+      },
+      dmnValues: { livingCategory: "SIH" },
+    } as WorkflowData;
+    // Update the fake data based on the options.
+    if (options?.hasDependants) {
+      fakeApplicationData.dependants = [
+        {
+          fullName: "Kid one",
+          dateOfBirth: "2017-12-06",
+          declaredOnTaxes: "yes",
+          attendingPostSecondarySchool: "no",
+        },
+      ];
+    }
+    if (options?.isLivingAway) {
+      workflowData.dmnValues.livingCategory = "SDA";
+    }
+    if (options?.isMarried) {
+      workflowData.studentData.relationshipStatus = RelationshipStatus.Married;
+    }
+    return saveFakeApplicationDisbursements(
+      db.dataSource,
+      {
+        student,
+        programYear: currentProgramYear,
+        firstDisbursementValues: [
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaLoan,
+            "CSLF",
+            100,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.BCGrant,
+            "BCSL",
+            100,
+          ),
+        ],
+      },
+      {
+        offeringIntensity: OfferingIntensity.fullTime,
+        applicationData: fakeApplicationData,
+        currentAssessmentInitialValues: {
+          assessmentData: fullTimeAssessmentData,
+          workflowData,
+          studentAssessmentStatus: StudentAssessmentStatus.Completed,
+        },
+      },
+    );
+  }
+
+  /**
+   * Create legacy application search data.
+   * @param individual legacy student.
+   * @param options related options.
+   * - `programYearId` program year id.
+   * - `institutionCode` whether the student is married.
+   * - `hasDependants` whether the student has dependants.
+   * @returns application search data.
+   */
+  async function createLegacyStudentApplicationSearchData(
+    individual: SFASIndividual,
+    options?: {
+      programYearId?: number;
```

Comment:
Having it as `programYearId` seems misleading at first. Can we have something that points directly to legacy, for instance, `sfasProgramYearId` or `legacyProgramYearId`?